docs/src/modules/components/AppLayoutDocs.js

```diff
@@ -119,21 +120,8 @@ export default function AppLayoutDocs(props) {
     throw new Error('Missing description in the page');
   }
 
-  const { canonicalAs } = pathnameToLanguage(router.asPath);
   let productName = 'MUI';
-  if (canonicalAs.startsWith('/material-ui/')) {
-    productName = 'Material UI';
-  } else if (canonicalAs.startsWith('/base-ui/')) {
-    productName = 'Base UI';
-  } else if (canonicalAs.startsWith('/x/')) {
-    productName = 'MUI X';
-  } else if (canonicalAs.startsWith('/system/')) {
-    productName = 'MUI System';
-  } else if (canonicalAs.startsWith('/toolpad/')) {
-    productName = 'MUI Toolpad';
-  } else if (canonicalAs.startsWith('/joy-ui/')) {
-    productName = 'Joy UI';
-  }
+  productName = convertProductIdToName(getProductInfoFromUrl(router.asPath));
```

Comment:
Much cleaner, but this does not support fall back to MUI. The `convertProductIdToName` will return `undefined` for unknown id.

```js
const productNameProductId = {
  'material-ui': 'Material UI',
  'joy-ui': 'Joy UI',
  'base-ui': 'Base UI',
  x: 'MUI X',
  system: 'MUI System',
  toolpad: 'Toolpad',
  'toolpad-studio': 'Toolpad Studio',
  'toolpad-core': 'Toolpad Core',
};

export function convertProductIdToName(productInfo) {
  return (
    productNameProductId[productInfo.productId] ||
    productNameProductId[productInfo.productCategoryId]
  );
}
```

Reply:
> but this does not support fall back to MUI.

Should it throw instead? Helping catch bugs sooner, I imagine we will never use MUI for a docs page.